pkg/operator/configobservation/apiserver/observer_apiserver.go

```diff
@@ -0,0 +1,199 @@
+package apiserver
+
+import (
+	"fmt"
+
+	"github.com/golang/glog"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+
+	"github.com/openshift/cluster-kube-apiserver-operator/pkg/operator/configobservation"
+	"github.com/openshift/cluster-kube-apiserver-operator/pkg/operator/operatorclient"
+	"github.com/openshift/library-go/pkg/operator/configobserver"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/resourcesynccontroller"
+)
+
+func ObserveUserClientCABundle(genericListers configobserver.Listers, recorder events.Recorder, existingConfig map[string]interface{}) (map[string]interface{}, []error) {
+	listers := genericListers.(configobservation.Listers)
+	var errs []error
+	previouslyObservedConfig := map[string]interface{}{}
+	observedConfig := map[string]interface{}{}
+
+	apiServer, err := listers.APIServerLister.Get("cluster")
+	if errors.IsNotFound(err) {
+		glog.Warningf("apiservers.config.openshift.io/cluster: not found")
+		return observedConfig, errs
+	}
+	if err != nil {
+		return previouslyObservedConfig, append(errs, err)
+	}
+
+	configMapName := apiServer.Spec.ClientCA.Name
+	if len(configMapName) == 0 {
+		return observedConfig, errs
```

Comment:
you forgot to clear the previous sync rule

You need to write unit tests

Reply:
Shouldn't you also return `previouslyObservedConfig` here?

Reply:
@soltysh `observedConfig` returned here because client ca is, in effect, being 'cleared'. That said, this observation func is odd in that it will always return an empty observed config map.

Reply:
@deads2k thanks. It appears that SyncConfigMap deletes destination on nil valued source?